src/App/components/Report/index.jsx

```diff
@@ -0,0 +1,53 @@
+import React from "react";
+import "./styles.css";
+import EmojiInput from "../EmojiInput/";
+
+const report = {
+  danger: [{ name: "obstacle", emoji: "🕳️" }, { name: "car", emoji: "🚘" }],
+  level: [
+    { name: "high", emoji: "😨" },
+    { name: "medium", emoji: "😵" },
+    { name: "low", emoji: "😒" },
+  ],
+};
+
+const gridFactory = (part, key, props) => {
+  const parts = {
+    grid: (t, i) => {
+      let cell = gridFactory("cell", key, props);
+      return cell(t, i);
+    },
+    cell: (t, i) => (
+      <EmojiInput
+        input={t}
+        parent={key}
+        selected={props[key]}
+        onClick={props.onClick}
+        key={`emojiInput-${t["name"]}`}
+      />
+    ),
+  };
+  return parts[part];
+};
+
+const Report = props => {
+  return (
+    <div>
```

Comment:
Is this `div` needed for styling or something?

Reply:
no, will remove.